packages/oauth/src/stores/file-store.ts

```diff
@@ -0,0 +1,116 @@
+  }
+
+  public async storeInstallation(installation: Installation, logger?: Logger): Promise<void> {
+    const { enterprise, team, user } = installation;
+    const installationData = JSON.stringify(installation);
+    const installationDir = this.getInstallationDir(enterprise?.id, team?.id);
+
+    if (logger !== undefined) { logger.warn('Storing installation in FileInstallationStore'); }
+
+    // Create Installation Directory
+    fs.mkdir(installationDir, { recursive: true }, (err) => {
+      if (err !== null) { return console.error(err); }
+    });
+
+    try {
+      writeToFile(`${installationDir}/app-latest`, installationData);
+      writeToFile(`${installationDir}/user-${user.id}-latest`, installationData);
+
+      if (this.historicalDataEnabled) {
+        const currentUTC = Date.now();
+        writeToFile(`${installationDir}/app-${currentUTC}`, installationData);
+        writeToFile(`${installationDir}/user-${user.id}-${currentUTC}`, installationData);
+      }
+    } catch (err) {
+      throw new Error('Failed to save installation to FileInstallationStore');
+    }
+  }
+
+  public async fetchInstallation(query: InstallationQuery<boolean>, logger?: Logger): Promise<Installation> {
+    const { enterpriseId, teamId } = query;
+    const installationDir = this.getInstallationDir(enterpriseId, teamId);
+
+    if (logger !== undefined) { logger.warn('Retrieving installation from FileInstallationStore'); }
+
+    try {
+      const data = fs.readFileSync(path.resolve(`${installationDir}/app-latest`));
+      const installation: Installation = JSON.parse(data.toString());
+      return installation;
+    } catch (err) {
+      throw new Error('Failed to fetch data from FileInstallationStore');
+    }
+  }
+
+  public async deleteInstallation(query: InstallationQuery<boolean>, logger?: Logger): Promise<void> {
+    const { enterpriseId, teamId, userId } = query;
+    const installationDir = this.getInstallationDir(enterpriseId, teamId);
+
+    if (logger !== undefined) { logger.warn('Deleting installation from FileInstallationStore'); }
+
+    let filesToDelete: string[] = ['app-latest'];
+
+    const appFiles = fs.readdirSync(installationDir).filter(file => file.includes('app-'));
+    filesToDelete = filesToDelete.concat(appFiles);
+
+    if (userId !== undefined) {
+      const userFiles = fs.readdirSync(installationDir).filter(file => file.includes(`user-${userId}-`));
+      filesToDelete = filesToDelete.concat(userFiles);
+    }
+
+    try {
+      filesToDelete.map(filePath => deleteFile(path.resolve(`${installationDir}/${filePath}`)));
+    } catch (err) {
+      throw new Error('Failed to delete installation from FileInstallationStore');
+    }
+  }
+
+  private getInstallationDir(enterpriseId = '', teamId = ''): string {
+    let installDir = `${this.baseDir}/${enterpriseId}`;
+
+    if (teamId !== '') {
+      installDir += (enterpriseId !== '') ? `-${teamId}` : `${teamId}`;
+    }
+
+    return installDir;
```

Comment:
Different to the Python implementation, installation directories are either a `enterprise_id-team_id`, `enterprise_id` or `team_id` format. I don't think this has any unintended consequences, but another point I specifically would like confirmation on.

Reply:
That's fine but the naming rules are not the same, I would recommend using a different base directory to avoid mixing different patterns in the same place. In other words, instead of `.bolt-app-installation`, you can use `.bolt-js-app-installation` or something similar. 

FYI, the Java SDK uses [`.slack-app`](https://github.com/slackapi/java-slack-sdk/blob/v1.8.1/bolt/src/main/java/com/slack/api/bolt/service/builtin/FileInstallationService.java#L24). We can revisit these naming if we would like to do so but for now, I think using different names would be fine.